src/bootstrap.ts

```diff
@@ -0,0 +1,400 @@
+    try {
+      const response = await fetch(
+        new URL(`things/bootstrap/${externalId}`, this.bootstrapsUrl).toString(), options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.bootstrapError.HandleError(errorRes.error, response.status)
+      }
+      const Bootstrap: Bootstrap = await response.json()
+      return Bootstrap
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async Bootstraps (queryParams: QueryParams, token: string): Promise<BootstrapPage> {
+    // Retrive all bootstraps with pagination
+    /**
+     * @method Bootstraps - Gets all bootstraps with pagination.
+     * @param {Object} queryParams - Query parameters.
+     * @param {String} token - Authentication token.
+     *  @returns {Object} - Bootstrap Page.
+     */
+    const stringParams: Record<string, string> = Object.fromEntries(
+      Object.entries(queryParams).map(([key, value]) => [key, String(value)])
+    )
+
+    const options: RequestInit = {
+      method: 'GET',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+    try {
+      const response = await fetch(
+        new URL(`things/config?${new URLSearchParams(stringParams).toString()}`, this.bootstrapsUrl).toString(), options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.bootstrapError.HandleError(errorRes.error, response.status)
+      }
+      const Bootstraps: BootstrapPage = await response.json()
+      return Bootstraps
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async UpdateBootstrapConnection (thingId: string, channels: Channel[], token: string): Promise<Response> {
+    // Update a bootstrap connection
+    /**
+    * @method UpdateConnection - Allows a logged in user to update the connection of a bootstrap configuration.
+    * @param {string} thing_id - The thing_id of the configuration to be updated.
+    * @param {object} channels - The channels object to be used for the update.
+    * @param {string} token - The token to be used for authentication.
+    * */
+    const options: RequestInit = {
+      method: 'PUT',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      },
+      body: JSON.stringify({ channels })
+    }
+    try {
+      const response = await fetch(
+        new URL(`things/configs/connections/${thingId}`, this.bootstrapsUrl).toString(), options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.bootstrapError.HandleError(errorRes.error, response.status)
+      }
+      const connResponse: Response = { status: response.status, message: 'Bootstrap Connection Successful' }
+      return connResponse
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async BootstrapEncrypt (inputBytes: Buffer, cryptoKey: string): Promise<string> {
```

Comment:
These two methods, `BoostrapEncrypt` and `BootstrapDecrypt` should be private methods only being used in the class. They should not be exported to be used outside the class

Reply:
They are being used in `BootstrapSecure` method